src/backend/BSUtil.pm

```diff
@@ -820,6 +820,24 @@ sub setdebuglevel {
   return $oldlevel;
 }
 
+=head2 openlog - open STDOUT/STDERR to log file
+
+ checks if $logfile is set and reopens STDOUT/STDERR to logfile
+
+ BSUtil::openlog($logfile, $user, $group);
+
+=cut
+
+sub openlog {
+  my ($logfile, $user, $group) = @_;
+  return unless defined $logfile;
+  $logfile = "$BSConfig::logdir/$logfile" unless $logfile =~ /\//;
```

Comment:
You can't have BSConfig references in BSUtil! Please move the logdir prepend code to the callers

Reply:
what about:
<pre>
 sub openlog {
-  my ($logfile, $user, $group) = @_;
+  my ($logfile, $logdir, $user, $group) = @_;
   return unless defined $logfile;
-  $logfile = "$BSConfig::logdir/$logfile" unless $logfile =~ /\//;
+  $logfile = "$logdir/$logfile" unless $logfile =~ /\//;
   my ($ld) = $logfile =~ m-(.*)/- ;
   BSUtil::mkdir_p_chown($ld, $user, $group) if $ld && defined($user) || defined($group);
   open(STDOUT, '>>', $logfile) || die("Could not open $logfile: $!\n");
</pre>
so that at least I don't have to copy the logic in dozen places ?